tests/testthat/test-variable-as-methods.R

```diff
@@ -0,0 +1,228 @@
+        
+        # alias to R's function names
+        expect_equal(as.numeric(ds$v1), as.Numeric(ds$v1))
+    })   
+
+    test_that("text to categorical", {
+        ds$v2_cat <- as.Categorical(ds$v2)
+        expect_true(is.derived(ds$v2_cat))
+        expect_true(is.Categorical(ds$v2_cat))
+        expect_equal(as.vector(ds$v2_cat), 
+                     as.factor(as.vector(ds$v2)))
+    })  
+    
+    test_that("datetime to numeric", {
+        ds$v5_num <- as.Numeric(ds$v5)
+        expect_true(is.derived(ds$v5_num))
+        expect_true(is.Numeric(ds$v5_num))
+        expect_equal(as.vector(ds$v5_num), 
+                     as.numeric(as.vector(ds$v5)))
+    }) 
+    
+    test_that("datetime to text", {
+        ds$v5_text <- as.Text(ds$v5)
+        expect_true(is.derived(ds$v5_text))
+        expect_true(is.Text(ds$v5_text))
+        expect_equal(as.vector(ds$v5_text), 
+                     format(as.vector(ds$v5), format = "%Y-%m-%d %H:%M:%S"))
+    }) 
+    
+    test_that("datetime to text", {
+        ds$v5_cat <- as.Categorical(ds$v5)
+        expect_true(is.derived(ds$v5_cat))
+        expect_true(is.Categorical(ds$v5_cat))
+        expect_equal(as.vector(ds$v5_cat), 
+                     as.factor(
+                         format(as.vector(ds$v5), format = "%Y-%m-%d %H:%M:%S")
+                     ))
+    }) 
+    
+    
+    test_that("text to datetime", {
+        ds$text_times <- format(as.vector(ds$v5), format = "%Y-%m-%d")
+        ds$times_from_text <- as.Datetime(ds$text_times, format = "%Y-%m-%d")
+        expect_true(is.derived(ds$times_from_text))
+        expect_true(is.Datetime(ds$times_from_text))
+        expect_equal(as.vector(ds$times_from_text), 
+                     as.vector(ds$v5))
+    }) 
+    
+    test_that("numeric to datetime", {
+        ds$num_times <- as.numeric(as.vector(ds$v5))
+        ds$times_from_num <- as.Datetime(ds$num_times, resolution = "D")
+        expect_true(is.derived(ds$times_from_num))
+        expect_true(is.Datetime(ds$times_from_num))
+        expect_equal(as.vector(ds$times_from_num), 
+                     as.vector(ds$v5))
+ 
+        # different resolutions and offsets change the result
+        ds$num_times_secs <- rep(365*24*60*60, 20)
+        ds$times_from_secs <- as.Datetime(ds$num_times_secs, resolution = "s",
+                                          offset = "1975-01-01")
+        expect_true(is.derived(ds$times_from_secs))
+        expect_true(is.Datetime(ds$times_from_secs))
+        expect_equal(as.vector(ds$times_from_secs), 
+                     as.POSIXlt(rep("1976-01-01", 20), tz = "UTC"))  
+        
+        ds$num_times_years <- rep(1, 20)
+        ds$times_from_years <- as.Datetime(ds$num_times_years, resolution = "Y")
+        expect_true(is.derived(ds$times_from_years))
+        expect_true(is.Datetime(ds$times_from_years))
+        # need rollup() here because as.vector doesn't currently support other rollups
```

Comment:
What's that? Are we missing something in as.vector?

Reply:
Yeah, currently rollups other than day, seconds, and milliseconds return NA in R. I'll make a separate issue that we can tackle separately. I need to dig into POSIX[cl]t to see if there are good ways to represent rolled up dates like that.